training/src/main/java/com/wolox/training/services/impl/UserServiceImpl.java

```diff
@@ -38,64 +38,47 @@ public void saveUser(User userInput) {
 
   @Override
   public User findUserById(Long id) throws UserNotFoundException {
-    Optional<User> user = this.userRepository.findById(id);
-    if (user.isPresent()) {
+    Optional<User> user = Optional.ofNullable(this.userRepository.findById(id)
+        .orElseThrow(() -> new UserNotFoundException("User not found", 404)));
```

Comment:
A good improvement to do with this code is to redefine a findById in the user repository, which returns an Optional<User> instead of a User. So you can avoid this Optional.ofNullable everytime you want to use this method, i mean, you will directly do: 
```
User user = this.userRepository.findById(id)
        .orElseThrow(() -> new UserNotFoundException("User not found", 404));
```

This comment applies to everytime you use userRepository.findById

To know more about Optionals you can use this [link](https://www.baeldung.com/java-optional)